lib/ansible/modules/windows/win_unzip.ps1

```diff
@@ -19,72 +19,81 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+# TODO: This module is not idempotent (it will always unzip and report change)
 
-$params = Parse-Args $args;
+$ErrorActionPreference = "Stop"
 
+$pcx_extensions = @('.bz2', '.gz', '.msu', '.tar', '.zip')
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$src = Get-AnsibleParam -obj $params -name "src" -type "path" -failifempty $true
+$dest = Get-AnsibleParam -obj $params -name "dest" -type "path" -failifempty $true
+$creates = Get-AnsibleParam -obj $params -name "creates" -type "path"
+$recurse = Get-AnsibleParam -obj $params -name "recurse" -type "bool" -default $false
+$delete_archive = Get-AnsibleParam -obj $params -name "delete_archive" -type "bool" -default $false -aliases 'rm'
+
+# Fixes a fail error message (when the task actually succeeds) for a
+# "Convert-ToJson: The converted JSON string is in bad format"
+# This happens when JSON is parsing a string that ends with a "\",
+# which is possible when specifying a directory to download to.
+# This catches that possible error, before assigning the JSON $result
 $result = @{
-    win_unzip = @{}
     changed = $false
+    dest = $dest -replace '\$',''
+    recurse = $recurse
+    removed = $false
+    src = $src -replace '\$',''
 }
 
-$creates = Get-AnsibleParam -obj $params -name "creates" -type "path"
-If ($creates -ne $null) {
-    If (Test-Path $creates) {
-        $result.msg = "The 'creates' file or directory ($creates) already exists."
-        Exit-Json $result
-    }
+If ($creates -and (Test-Path -LiteralPath $creates)) {
+    $result.skipped = $true
+    Exit-Json -obj $result -message "The file or directory '$creates' already exists."
```

Comment:
Then we have a wider problem, because the following modules are doing it too:
- win_domain_controller
- win_msi
- win_uri
- win_timezone
- win_acl